src/apps/backend/modules/password-reset-token/password-reset-token-service.ts

```diff
@@ -0,0 +1,124 @@
+import { EmailService } from '../communication';
+import { SendEmailParams } from '../communication/types';
+import { ConfigService } from '../config';
+
+import PasswordResetTokenReader from './internal/password-reset-token-reader';
+import PasswordResetTokenUtil from './internal/password-reset-token-util';
+import PasswordResetTokenWriter from './internal/password-reset-token-writer';
+import { CreatePasswordResetTokenParams, PasswordResetToken, PasswordResetTokenEmailNotEnabledForTheEnvironmentError } from './types';
+
+export default class PasswordResetTokenService {
+  public static async createPasswordResetToken(
+    params: CreatePasswordResetTokenParams,
+  ): Promise<PasswordResetToken> {
+    const account = await AccountService.getAccountByUsername(params.username);
+
+    const token = PasswordResetTokenUtil.generatePasswordResetToken();
+
+    const passwordResetToken = await PasswordResetTokenWriter.createPasswordResetToken(
+      account.id,
+      token,
+    );
+
+    await this.sendPasswordResetEmail(
+      account.id,
+      account.firstName,
+      account.username,
+      token,
+    );
+
+    return passwordResetToken;
+  }
+
+  public static async getPasswordResetTokenByAccountId(
+    accountId: string,
+  ): Promise<PasswordResetToken> {
+    return PasswordResetTokenReader.getPasswordResetTokenByAccountId(accountId);
+  }
+
+  public static async setPasswordResetTokenAsUsedById(
+    passwordResetTokenId: string,
+  ): Promise<PasswordResetToken> {
+    return PasswordResetTokenWriter.setPasswordResetTokenAsUsed(
+      passwordResetTokenId,
+    );
+  }
+
+  public static async verifyPasswordResetToken(
+    accountId: string,
+    token: string,
+  ): Promise<PasswordResetToken> {
+    const passwordResetToken = await PasswordResetTokenService
+      .getPasswordResetTokenByAccountId(accountId);
+
+    if (passwordResetToken.isExpired) {
+      throw new AccountBadRequestError(
+        `Password reset link is expired for accountId ${accountId}. Please retry with new link`,
+      );
+    }
+
+    if (passwordResetToken.isUsed) {
+      throw new AccountBadRequestError(
+        `Password reset is already used for accountId ${accountId}. Please retry with new link`,
+      );
+    }
+
+    const isTokenValid = await PasswordResetTokenUtil.comparePasswordResetToken(
+      token,
+      passwordResetToken.token,
+    );
+    if (!isTokenValid) {
+      throw new AccountBadRequestError(
+        `Password reset link is invalid for accountId ${accountId}. Please retry with new link.`,
+      );
+    }
+
+    return passwordResetToken;
+  }
+
+  private static async sendPasswordResetEmail(
+    accountId: string,
+    firstName: string,
+    username: string,
+    passwordResetToken: string,
+  ): Promise<void> {
+    const passwordResetEmailEnabled = JSON.parse(ConfigService.getValue<string>(
+      'accounts.passwordResetEmailEnabled',
+    )) as boolean;
```

Comment:
This is going to try to typecast string `PASSWORD_RESET_EMAIL_ENABLED` to boolean which seems wrong also. Don't you have `ConfigService.getValue<boolean>`?